code/modules/mob/living/simple_animal/hostile/ordeal/indigo.dm

```diff
@@ -45,3 +45,127 @@
 	adjustBruteLoss(-(maxHealth/2))
 	L.gib()
 	return TRUE
+
+
+/mob/living/simple_animal/hostile/ordeal/indigo_midnight
+	name = "The Matriarch"
+	desc = "A humanoid creature wearing metallic armor. The Queen of sweepers."
+	icon = 'ModularTegustation/Teguicons/64x64.dmi'
+	icon_state = "matriarch"
+	icon_living = "matriarch"
+	icon_dead = "matriarch_dead"
+	faction = list("indigo_ordeal")
+	maxHealth = 2000
+	health = 2000
+	stat_attack = DEAD
+	pixel_x = -16
+	base_pixel_x = -16
+	melee_damage_type = BLACK_DAMAGE
+	armortype = BLACK_DAMAGE
+	rapid_melee = 2
+	melee_damage_lower = 60
+	melee_damage_upper = 60
+	attack_verb_continuous = "stabs"
+	attack_verb_simple = "stab"
+	attack_sound = 'sound/effects/ordeals/indigo/stab_1.ogg'
+	damage_coeff = list(BRUTE = 0.2, RED_DAMAGE = 0.3, WHITE_DAMAGE = 0.4, BLACK_DAMAGE = 0.2, PALE_DAMAGE = 0.05)
+	blood_volume = BLOOD_VOLUME_NORMAL
+	move_resist = MOVE_FORCE_OVERPOWERING
+
+	//How many people has she eaten
+	var/belly = 0
+
+	var/pulse_cooldown
+	var/pulse_cooldown_time = 10 SECONDS
+	var/pulse_damage = 10 // More over time
+
+
+/mob/living/simple_animal/hostile/ordeal/indigo_midnight/AttackingTarget()
+	. = ..()
+	if(. && isliving(target))
+		var/mob/living/L = target
+		if(L.stat != DEAD)
+			if(L.health <= HEALTH_THRESHOLD_DEAD && HAS_TRAIT(L, TRAIT_NODEATH))
+				devour(L)
+		else
+			devour(L)
+
+
+/mob/living/simple_animal/hostile/ordeal/indigo_midnight/proc/devour(mob/living/L)
+	if(!L)
+		return FALSE
+	visible_message(
+		"<span class='danger'>[src] devours [L]!</span>",
+		"<span class='userdanger'>You feast on [L], restoring your health!</span>")
+	adjustBruteLoss(-(maxHealth))
+	L.gib()
+
+
+	//Increase the Vore counter by 1
+	belly += 1
+	pulse_damage += 2
+
+
+	//She gets faster but not as protected or as strong
+	if(belly == 5)
+		//animation!
+		icon_state = "striptease"
```

Comment:
What

Reply:
Ignore my shitty icon naming I didn't push the fix